### PR TITLE
shell: Take docs from the correct manifest item

### LINKED ISSUE #21040: shell: Doesn't show "network" docs when on "network/firewall" page
When on the "network/firewall" page, the "Help" menu does not include the networking docs:

image

### PR DESCRIPTION
**Pull request by mvollmer**
Fixes #21040

### CODE REVIEW

**Comment by mvollmer** [comment]
One can argue that "network/firefox" should have its own list of documentation links. We would have to ivent something new for our manifests for that, such as a new "docs" top-level section:

```
{
  "menu": {
    "index":  {
      "docs": [ /* for `"network"` */ ]
   }
 },

 "docs": {
    "firewall": [ /* for "/network/firewall" */ ],
    "": [ /* for "/network" */ ],
  }
}
```

We could also use that instead of `{ "parent": { "docs": ... } }`.

File: pkg/shell/util.jsx [hunk 1]
```diff
@@ -168,20 +168,27 @@ function CompiledComponents() {
...
         }
 
         return self.items[component] || { path, label: path, section: "menu" };
     };
+
+    self.find_path_manifest = function(path) {
+        const parts = path.split("/");
+        const pkg = parts[0];
+
+        return manifests[pkg] || { };
```
Comment (cockpituous) [inline pkg/shell/util.jsx:183]: This added line is not executed by any test.

**Comment by mvollmer** [comment]
> One can argue [...]

But whatever the result of that argument is, I think sub-pages should show the docs of their manifest menu entry by default, instead of nothing. So we should have this bug fix in any case.

**Review by martinpitt — APPROVED** [review]
Thanks! After the Matrix discussion I now tend towards "let's show all of them on all subpages" (i.e. what this PR does).

It's not clear to me whether the firewall case generalizes, i.e. if you'd really never want to show the parent pages for a subpage. There are not so many doc pages that the menu would become too crowded, and I think "subpage docs only" would need more research to justify the manifest definition change. (That's not a "no", I just don't think it has been done, or is wort ith)

**Comment by mvollmer** [comment]
> There are not so many doc pages that the menu would become too crowded, and I think "subpage docs only" would need more research to justify the manifest definition change. (That's not a "no", I just don't think it has been done, or is wort ith)

What we want is probably "additional docs for sub-pages". E.g., /network/firewall shows the docs of /network plus firewall specific docs.

But then again, /network might want to show docs that are not applicable to teh firewall (bonds, bridges, ...). So just having separate lists of docs for each sub-page in the manifest seems most flexible. We can manually make them share whatever seems right.

**Pull request merged**